src/libopensc/pkcs15-iasecc.c

```diff
@@ -103,6 +103,44 @@ _iasecc_md_update_keyinfo(struct sc_pkcs15_card *p15card, struct sc_pkcs15_objec
 }
 
 
+/*
+ * CPx cards have an undocumented issue: they lack of
+ * Algorithm's reference into their PKCS's ASN1 encoding.
+ */
+static int
+_iasecc_cpx_fixup_prkdf(struct sc_pkcs15_card *p15card)
+{
+	struct sc_context * const ctx = p15card->card->ctx;
+	int rv = SC_SUCCESS;
+#ifdef ENABLE_OPENSSL
+	struct sc_pkcs15_object *pkobjs[32];
+	int ii, count;
+
+	LOG_FUNC_CALLED(ctx);
+
+	rv = sc_pkcs15_get_objects(p15card, SC_PKCS15_TYPE_PRKEY, pkobjs, sizeof(pkobjs)/sizeof(pkobjs[0]));
+	LOG_TEST_RET(ctx, rv, "Cannot get PRKEY objects list");
+
+	count = rv;
+	for(ii=0; ii<count; ii++)   {
+		rv = iasecc_pkcs15_encode_supported_algos(p15card, pkobjs[ii]);
+		LOG_TEST_RET(ctx, rv, "Cannot fix suported_algos");
+	}
+
+	LOG_FUNC_RETURN(ctx, rv);
+#else
+	LOG_FUNC_CALLED(ctx);
+	sc_log(ctx, "OpenSC was built without OpenSSL support: skipping");
+	LOG_FUNC_RETURN(ctx, rv);
```

Comment:
one more note, sorry for not noticing on the first sight. You can move the `LOG_FUNC_RETURN` out of the ifdef to avoid code duplication. as it is the same in both branches.

Reply:
done, I could have guessed it myself ;)